### PR TITLE
Fixes bug in leafletPathEvents

### LINKED ISSUE #728: Labels broken for paths
Markers and Paths use leafletLabelEvents.genLabelEvents differently.

In leafletMarkerEvents (line 169):
```javascript
            if (Helpers.LabelPlugin.isLoaded() && isDefined(lObject.label)) {
                lblHelp.genLabelEvents(name, logic, leafletScope, lObject, model, layerName);
            }
```
In leafletPathEvents (line 114):   **BROKEN**
```javascript
        if (Helpers.LabelPlugin.isLoaded() && isDefined(lObject.label)) {
            lblHelp.genLabelEvents(leafletScope, logic, lObject, name);
        }
```

leafletPathEvents should follow the function definition in leafletLabelEvents:
```javascript
    var _genLabelEvents = function (eventName, logic, leafletScope, lObject, name, model, layerName) {
       ...
    }
```

### PR DESCRIPTION
**Pull request by pieterjandesmedt**
Fixes #728 and #729

### CODE REVIEW

File: src/directives/geojson.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ angular.module("leaflet-directive")
                         onEachFeature = geojson.onEachFeature;
                     } else {
                         onEachFeature = function(feature, layer) {
-                            if (leafletHelpers.LabelPlugin.isLoaded() && isDefined(geojson.label)) {
+                            if (leafletHelpers.LabelPlugin.isLoaded() && isDefined(feature.properties.description)) {
```
Comment (nmccready) [inline src/directives/geojson.js]: Should properties itself be optional? So 
```
if (leafletHelpers.LabelPlugin.isLoaded() && isDefined(feature.properties) && isDefined(feature.properties.description))
```

**Comment by nmccready** [comment]
Nice catch, so I guess there was in recursive watch loop there. BTW both watches should really check against unnecessary changes.
```
leafletScope.$watch('layers.overlays', function(newOverlayLayers, old) {
if(angular.equals(newOverlaysLayer, old)) return;
});
```

**Comment by nmccready** [comment]
Can you add in the watch checks as well?

File: src/directives/geojson.js (hunk 1 above)
Comment (pieterjandesmedt) [inline src/directives/geojson.js]: According to the [geojson spec](http://geojson.org/geojson-spec.html#feature-objects), a feature should always have properties, but I guess not all geojsons in the wild adhere to that spec.

So yeah, adding `isDefined(feature.properties)` is a good idea.

**Comment by pieterjandesmedt** [comment]
Are you sure it should double check the watch? Isn't that the definition of a watch that it only fires when old and new values are different?

**Comment by nmccready** [comment]
I guess we can skip it, but I have seen them fired a lot when nothing changes. They also always run on the first check regardless.

**Comment by nmccready** [comment]
@tombatossals looks like you have another PR that conflicts with this. #734 can you jump in here?

**Comment by pieterjandesmedt** [comment]
Added the check. We can always remove it if necessary.

**Comment by nmccready** [comment]
@pieterjandesmedt LGTM, but I am waiting on @tombatossals since he has #734

**Comment by tombatossals** [comment]
Hey guys, I'm merging this and will solve the conflics in the https://github.com/tombatossals/angular-leaflet-directive/pull/734

Nice work

**Pull request merged**